use-shopping-cart/core/warning-middleware.js

```diff
@@ -0,0 +1,20 @@
+function checkCountType(count, action) {
+  if (typeof count !== 'number') {
+    console.warn(`${action.type} requires that count be a number`, action)
+  }
+}
```

Comment:
I would recommend changing this to handle both `typeof count` and `> 0` and also letting it return whether or not the `count` was a valid one (`true` or `false`).

```js
function isValidCount(count, action) {
  if (typeof count !== 'number') {
    console.warn(/* ... */)
    return false // Invalid count
  }
  if (count <= 0) {
    console.warn(/* ... */)
    return false // Invalid count
  }
  // Valid count
  return true
}
```

Reply:
I like this!